multi-tenant: Use unrestricted blob client for MT tests

ajstorm opened the pull request:
See individual commits for details.

Informs #76378.

Release note: None.

ajstorm committed:
multi-tenant: Use unrestricted blob client for MT tests

Previously we were using a local-only blob client factory for the multi-tenant
tests. This caused test case failures however, because a local-only blob client
can only access storage on node 0. This change uses a unrestricted blob client
so remote node storage can be accessed.

Informs: #76378

Release note: None.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/84200)
<!-- Reviewable:end -->

ajstorm committed:
bulk: Enable some tests which fail under default tenant

Some tests which were not safe to be run under the default test tenant can now
be enabled to run in that mode. The change required was to differentiate access
to the system and secondary tenants in a couple of places. One other test case
could be enabled without any additional changes. In a couple of other cases, we
adjusted the comments to make it clear _why_ the test could not run under the
default test tenant.

Informs: #76378

Release note: None

knz reviewed (approved):
> Reviewed 2 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84200)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @adityamaru and @renatolabs)



<!-- Sent from Reviewable.io -->

adityamaru commented on pkg/server/testserver.go:
```diff
@@ -817,11 +817,14 @@ func (ts *TestServer) StartTenant(
-	//  working for the common multi-tenant case. Will need to revisit this
-	//  to enable tests which require remote blob access. Tracked with #76378.
+	localNodeIDContainer := &base.NodeIDContainer{}
+	localNodeIDContainer.Set(context.Background(), ts.NodeID())
```
nit: can we just use the passed in `ctx` here?

ajstorm replied:
Done.  TFTR!

ajstorm commented:
TFTRs!

bors r=knz,adityamaru
